app/components/treatment/Treatment.js

```diff
@@ -21,7 +21,15 @@ export class Treatment extends Component {
 
   render() {
 
-    const { treatment } = this.props
+    const { plan } = this.props
+    const { treatments } = plan
+    const treatmentId = Number(this.props.params.treatmentId)
```

Comment:
Hmm I like that. Then maybe we don't need a treatment or ID on the store at all. We can simply use the params and the `plan` on the store to select the treatment we are interested in off the state in `mapStateToProps`